collector/src/descriptor/kind/bridge_network_status.rs

```diff
@@ -0,0 +1,199 @@
+            |acc, line| -> Result<NetworkStatusBuilder, Error> {
+                match parse_line(line) {
+                    ("r", params) => {
+                        let mut builder = acc?;
+
+                        if let Ok(net) = builder.build() {
+                            network_status.push(net);
+                            builder = NetworkStatusBuilder::default();
+                        }
+
+                        Ok(builder
+                            .nickname(params[0].to_string())
+                            .identity(params[1].to_string())
+                            .digest(params[2].to_string())
+                            .publication(date(&format!("{} {}", params[3], params[4]))?.1)
+                            .ipv4(params[5].parse()?)
+                            .or_port(params[6].parse()?)
+                            .dir_port(params[7].parse()?)
+                            .to_owned())
+                    }
+                    ("a", params) => {
+                        let builder = acc?;
+                        Ok(builder.addresses(Address::parse(params[0])?))
+                    }
+                    ("s", params) => {
+                        let mut builder = acc?;
+                        Ok(builder
+                            .flags(params.iter().map(|elem| elem.to_string()).collect())
+                            .to_owned())
+                    }
+                    ("w", params) => {
+                        let mut builder = acc?;
+                        Ok(builder
+                            .bandwidth(
+                                params[0]
+                                    .split_once('=')
+                                    .ok_or_else(|| {
+                                        ErrorKind::MalformedDesc("Bandwidth malformed".to_owned())
+                                    })?
+                                    .1
+                                    .parse()?,
+                            )
+                            .to_owned())
+                    }
+                    ("p", params) => {
+                        let builder = acc?;
+                        let pol = match params[0] {
+                            "accept" => Policy::Accept(params[1].to_owned()),
+                            "reject" => Policy::Reject(params[1].to_owned()),
+                            _ => unreachable!(),
+                        };
+                        Ok(builder.policies(pol))
+                    }
+                    // handle empty line
+                    ("", _) => acc,
+                    (_any, _value) => {
+                        unreachable!()
+                    }
+                }
+            },
+        )?;
+
+        Ok(BridgeNetworkStatus {
+            header,
+            network_status,
+        })
+    }
+}
+
+fn parse_line(input: &str) -> (&str, Vec<&str>) {
+    let t = input.split(' ').collect::<Vec<&str>>();
+    (t[0], t[1..].to_vec())
```

Comment:
this could panic on invalid document. You can use `t.remove(0)` and check for `None`, it should also avoid creating a 2nd `Vec` (though that might be optimized away anyway)